### PR TITLE
Adds the single action revolver. Probably.

### PR DESCRIPTION
**Pull request by SomeguyManperson**
# Document the changes in your pull request

Goofy-assed and SOURCE ACCURATE single action revolver, firing and reloading are gimmicky and I haven't exactly decided where to put the damn thing but I need feedback so

# Spriting

# Wiki Documentation

tbd

# Changelog

:cl:  Theos, biome for sprites 
rscadd: New gun?? Reloading mechanic??? gimmicky as hell??
imageadd: new gun
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 1]
```diff
@@ -154,6 +154,121 @@
...
+		return
+	if(hammer_set == HAMMER_FULL || (ismob(target) && user.a_intent == INTENT_GRAB)) //able to shoot or attempting a holdup
+		return ..()
+	to_chat(user, span_warning("You pull back [src]'s hammer."))
+	playsound(src, dry_fire_sound, 30, TRUE)
+	user.changeNext_move(CLICK_CD_RANGE)
+	hammer_set = HAMMER_FULL
+	cut_overlays()
+	flick("single-action_full_anim", src)
+	addtimer(CALLBACK(src, .proc/update_icon), 3)
```
Comment (tattax) [inline code/modules/projectiles/guns/ballistic/revolver.dm:213]:
```suggestion
	addtimer(CALLBACK(src, PROC_REF(update_icon)), 3)
```
this is gonna be needed soon after 515 update

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 2]
```diff
@@ -154,6 +154,121 @@
...
+		user.changeNext_move(CLICK_CD_RANGE)
+		if(casing && !casing.BB)
+			casing.forceMove(drop_location())
+			magazine.get_round(FALSE, FALSE) //this removes the round from our cylinder
+			to_chat(user, span_warning("You push [casing] out of [src]."))
+			return
+		to_chat(user, span_warning("You rotate [src]'s cylinder."))
+		cylinder.rotate()
+		casing = magazine.get_round(TRUE, FALSE)
+		balloon_alert(user, casing ? casing.BB ? "Chamber loaded" : "Chamber spent" : "Chamber empty")
```
Comment (tattax) [inline code/modules/projectiles/guns/ballistic/revolver.dm:203]: balloon alerts normally have the first letter not capitalized, unless it's something specific, just want to remain consistent

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 3]
```diff
@@ -154,6 +154,121 @@
...
+	playsound(src, dry_fire_sound, 30, TRUE)
+	user.changeNext_move(CLICK_CD_RANGE)
+	hammer_set = HAMMER_FULL
+	cut_overlays()
+	flick("single-action_full_anim", src)
+	addtimer(CALLBACK(src, .proc/update_icon), 3)
+
+/obj/item/gun/ballistic/revolver/single_action/attack_self(mob/user)
+	if(hammer_set == HAMMER_RELOAD)
+		balloon_alert(user, "Now aiming.")
```
Comment (tattax) [inline code/modules/projectiles/guns/ballistic/revolver.dm:217]: same thing with single dots

File: code/modules/projectiles/guns/ballistic/revolver.dm [hunk 4]
```diff
@@ -154,6 +154,121 @@
...
+		playsound(src, dry_fire_sound, 30, TRUE)
+		hammer_set = HAMMER_FULL
+		chamber_round(FALSE)
+		update_icon()
+	else
+		balloon_alert(user, "Now reloading.")
+		to_chat(user, span_warning("You pull [src]'s hammer halfway back and begin reloading."))
+		hammer_set = HAMMER_RELOAD
+		chambered = null
+		update_icon()
```
Comment (tattax) [inline code/modules/projectiles/guns/ballistic/revolver.dm:229]:
```suggestion
	else
		balloon_alert(user, "Now reloading.")
		to_chat(user, span_warning("You pull [src]'s hammer halfway back and begin reloading."))
		hammer_set = HAMMER_RELOAD
		chambered = null
	update_icon()
```

**Comment by SomeguyManperson** [comment]
needs work

**Pull request closed without merging**